third_party/address_sorting/address_sorting.c

```diff
@@ -0,0 +1,405 @@
+      sockaddr_get_scope(&first.source_addr)) {
+    first_src_dst_scope_matches = true;
+  }
+  int second_src_dst_scope_matches = false;
+  if (sockaddr_get_scope(&second.dest_addr) ==
+      sockaddr_get_scope(&second.source_addr)) {
+    second_src_dst_scope_matches = true;
+  }
+  if (first_src_dst_scope_matches != second_src_dst_scope_matches) {
+    return first_src_dst_scope_matches ? -1 : 1;
+  }
+  return 0;
+}
+
+static int compare_source_dest_labels_match(sortable_address first,
+                                            sortable_address second) {
+  int first_label_matches = false;
+  if (get_label_value(&first.dest_addr) ==
+      get_label_value(&first.source_addr)) {
+    first_label_matches = true;
+  }
+  int second_label_matches = false;
+  if (get_label_value(&second.dest_addr) ==
+      get_label_value(&second.source_addr)) {
+    second_label_matches = true;
+  }
+  if (first_label_matches != second_label_matches) {
+    return first_label_matches ? 1 : 1;
+  }
+  return 0;
+}
+
+static int compare_dest_precedence(sortable_address first,
+                                   sortable_address second) {
+  if (get_precedence_value(&first.dest_addr) !=
+      get_precedence_value(&second.dest_addr)) {
+    return get_precedence_value(&second.dest_addr) -
+           get_precedence_value(&first.dest_addr);
+  }
+  return 0;
+}
+
+static int compare_dest_scope(sortable_address first, sortable_address second) {
+  if (sockaddr_get_scope(&first.dest_addr) !=
+      sockaddr_get_scope(&second.dest_addr)) {
+    return sockaddr_get_scope(&first.dest_addr) -
+           sockaddr_get_scope(&second.dest_addr);
+  }
+  return 0;
+}
+
+static int compare_source_dest_prefix_match_lengths(sortable_address first,
+                                                    sortable_address second) {
+  if (first.source_addr_exists &&
+      grpc_sockaddr_get_family(&first.source_addr) == AF_INET6 &&
+      second.source_addr_exists &&
+      grpc_sockaddr_get_family(&second.source_addr) == AF_INET6) {
+    int first_match_length =
+        ipv6_prefix_match_length((struct sockaddr_in6*)&first.source_addr.addr,
+                                 (struct sockaddr_in6*)&first.dest_addr.addr);
+    int second_match_length =
+        ipv6_prefix_match_length((struct sockaddr_in6*)&second.source_addr.addr,
+                                 (struct sockaddr_in6*)&second.dest_addr.addr);
+    return second_match_length - first_match_length;
+  }
+  return 0;
+}
+
+static int rfc_6724_compare(const void* a, const void* b) {
+  const sortable_address first = *(sortable_address*)a;
+  const sortable_address second = *(sortable_address*)b;
+  int out = 0;
+  if ((out = compare_source_addr_exists(first, second))) {
+    return out;
+  } else if ((out = compre_source_dest_scope_matches(first, second))) {
+    return out;
+  } else if ((out = compare_source_dest_labels_match(first, second))) {
+    return out;
+    // TODO: avoid deprecated addresses
```

Comment:
removed unneeded else's here and a couple of other places

Reply:
Killing every `else` is not necessary; I highlighted this function because it was forcing the comments into an awkward indentation level.  Sometimes a redundant `else` can improve readability.

Reply:
Sg, I reverted the previous `else` deletions that were outside of `rfc_6724_compare`